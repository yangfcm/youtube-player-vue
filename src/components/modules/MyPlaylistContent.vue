<template>
  <div>
    <app-loader v-if="!playlists && !error"></app-loader>
    <app-error-message v-if="error">{{ error }}</app-error-message>
    <div v-if="playlists && !error">
      <app-page-title>My Playlist</app-page-title>
      <app-playlist-grid :playlists="playlists.items"></app-playlist-grid>
      <app-blank></app-blank>
      <app-more-button>
        :nextPageToken="playlists.nextPageToken"
        :isLoadingMore = "isLoadingMore"
        @onClickMore = 'handleMore($event)'
      </app-more-button>
    </div>
  </div>
</template>

<script>
import { mapState, mapActions, mapGetters } from "vuex";
import Loader from "../common/Loader";
import ErrorMessage from "../common/ErrorMessage";
import PageTitle from "../common/PageTitle";
import MoreButton from "./MoreButton";
import PlaylistGrid from "./PlaylistGrid";
import Blank from "../common/Blank";

export default {
  components: {
    appLoader: Loader,
    appErrorMessage: ErrorMessage,
    appPageTitle: PageTitle,
    appMoreButton: MoreButton,
    appPlaylistGrid: PlaylistGrid,
    appBlank: Blank,
  },
  data: function () {
    return {
      playlists: null,
      error: "",
      isLoadingMore: false,
    };
  },
  computed: {
    ...mapState(["playlist"]),
    ...mapGetters(["playlistErrorMessage"]),
  },
  methods: {
    ...mapActions(["fetchMyPlaylist"]),
    handleMore($event) {
      console.log("handle more");
    },
  },
  async created() {
    await this.fetchMyPlaylist();
    if (this.playlist.playlistError) {
      this.error = this.playlistErrorMessage;
    } else if (this.playlist.myPlaylist) {
      this.playlists = this.playlist.myPlaylist;
      this.error = "";
    }
  },
};
</script>

<style scoped>
</style>