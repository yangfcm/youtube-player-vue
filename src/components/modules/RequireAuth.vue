<template>
  <div class="ui red message">
    <app-blank></app-blank>
    <div class="ui header aligned center">You have not signed in</div>
    <app-blank></app-blank>
    <div class="ui header aligned center">
      <button class="ui youtube button negative basic" @click="handleSignIn">
        <i class="youtube icon"></i>
        Sign in
      </button>
    </div>
    <app-blank></app-blank>
  </div>
</template>

<script>
import googleAuth from "../../mixins/googleAuth";
import Blank from "../common/Blank";
export default {
  components: {
    appBlank: Blank,
  },
  mixins: [googleAuth],
  methods: {
    handleSignIn() {
      this.handleGoogleSignin();
    },
  },
};
</script>

<style scoped>
</style>