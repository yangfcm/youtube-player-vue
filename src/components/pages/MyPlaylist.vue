<template>
  <!-- MyPlaylist page - Playlist for authed user -->
  <div>
    <app-user-banner></app-user-banner>
    <app-menu></app-menu>
  </div>
</template>

<script>
import UserBanner from "../layout/UserBanner";
import Menu from "../layout/Menu";
import Loader from "../common/Loader";
import ErrorMessage from "../common/ErrorMessage";

export default {
  components: {
    appMenu: Menu,
    appUserBanner: UserBanner,
    appLoader: Loader,
    appErrorMessage: ErrorMessage,
  },
};
</script>

<style scoped></style>
