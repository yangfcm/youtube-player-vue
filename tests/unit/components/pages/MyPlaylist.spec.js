import { shallowMount } from "@vue/test-utils";
import MyPlaylist from "@/components/pages/MyPlaylist";
import RequireAuth from "@/components/modules/RequireAuth";
import MyPlaylistContent from "@/components/modules/MyPlaylistContent";

describe("Test MyPlaylist page", () => {
  test("should show RequireAuth component if user does not log in", () => {
    const $store = {
      getters: {
        signedIn: false,
      },
    };
    const wrapper = shallowMount(MyPlaylist, {
      mocks: {
        $store,
      },
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.findComponent(RequireAuth).exists()).toBe(true);
    expect(wrapper.findComponent(MyPlaylistContent).exists()).toBe(false);
  });

  test("should show MyPlaylistContent component if user logs in", () => {
    const $store = {
      getters: {
        signedIn: true,
      },
    };
    const wrapper = shallowMount(MyPlaylist, {
      mocks: {
        $store,
      },
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.findComponent(RequireAuth).exists()).toBe(false);
    expect(wrapper.findComponent(MyPlaylistContent).exists()).toBe(true);
  });
});
