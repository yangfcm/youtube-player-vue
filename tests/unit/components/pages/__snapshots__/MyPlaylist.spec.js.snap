// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyPlaylist page should show MyPlaylistContent component if user logs in 1`] = `
<div>
  <app-user-banner-stub></app-user-banner-stub>
  <app-menu-stub></app-menu-stub>
  <app-my-playlist-content-stub></app-my-playlist-content-stub>
  <!---->
</div>
`;

exports[`Test MyPlaylist page should show RequireAuth component if user does not log in 1`] = `
<div>
  <app-user-banner-stub></app-user-banner-stub>
  <app-menu-stub></app-menu-stub>
  <!---->
  <app-require-auth-stub></app-require-auth-stub>
</div>
`;
